### ISSUE TITLE
Kernel Crash with MultiRMSE and Lossguide or Depthwise

### ISSUE THREAD

**Issue by ThomasWolf0701** [catboost/catboost#1515]
Problem: When running MultiRMSE with Lossguide or Depthwise I get a kernel crash on the cpu
With Symmetric Trees it runs fine. The same for normal non multilabel regression and Lossguide

bestModel =  classifiers[1].fit(scipy.sparse.csr_matrix(featureMatrix.values),np.array(values))
Pass groups=None as keyword args. From version 0.25 passing these as positional arguments will result in an error
[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.
Starting bracket 1 (out of 3) of hyperband
Starting successive halving iteration 1 out of 3. Fitting 9 configurations, with resource_param m__iterations set to 11, and keeping the best 3 configurations.
Fitting 3 folds for each of 9 candidates, totalling 27 fits
[CV] f__f__max_features=200, m__bagging_temperature=0.9507143064099162, m__depth=11, m__grow_policy=Lossguide, m__l2_leaf_reg=72, m__random_strength=0.0009695826644515221, m__iterations=11 
0:	learn: 7.9005190	total: 3.08ms	remaining: 30.9ms
1:	learn: 7.8269764	total: 4.5ms	remaining: 20.3ms
2:	learn: 7.7558020	total: 5.48ms	remaining: 14.6ms
3:	learn: 7.6869345	total: 6.57ms	remaining: 11.5ms
4:	learn: 7.6203134	total: 7.49ms	remaining: 8.98ms
5:	learn: 7.5558793	total: 8.32ms	remaining: 6.93ms
6:	learn: 7.4935735	total: 9.48ms	remaining: 5.42ms
7:	learn: 7.4333385	total: 10.7ms	remaining: 4ms
8:	learn: 7.3751175	total: 11.6ms	remaining: 2.58ms
9:	learn: 7.3188549	total: 12.5ms	remaining: 1.25ms
10:	learn: 7.2644958	total: 13.3ms	remaining: 0us

Restarting kernel... 
 
[SpyderKernelApp] WARNING | No such comm: cba83eba3ca511eb87119cb6d01cd292

catboost version: 0.24.3
Operating System: Windows 10
CPU: Intel Core i7
# GPU:

**Comment by ek-ak** [comment]
Hello! Please give the minimal reproducing example.

**Comment by ThomasWolf0701** [comment]
> 
> 
> Hello! Please give the minimal reproducing example.

I attached the data as a zip and here is the code that reproduces the crash:
import pandas as pd
from catboost import CatBoostRegressor


values = pd.read_pickle("multivalues.pkl")
featureMatrix = pd.read_pickle("allfeatures.pkl")

cat_model = CatBoostRegressor(iterations = 11,grow_policy = "Lossguide",objective = "MultiRMSE",
                              bagging_temperature=0.9507143064099162, depth=11,l2_leaf_reg=72, random_strength=0.0009695826644515221)

cat_model.fit(featureMatrix,values)


[multibug.zip](https://github.com/catboost/catboost/files/5698618/multibug.zip)

**Comment by ThomasWolf0701** [comment]
When i remove random_strength or set it to 1 the code actually works.

**Comment by ThomasWolf0701** [comment]
When setting the random_strength to a value equal to or larger than one the kernel does not crash. But then if i start to tune the learning rate I still get a kernel crash from time to time for Depthwise and Lossguide

**Comment by ThomasWolf0701** [comment]
Now it also crashes sometimes without setting random_strength or or the learning rate:

[CV]  f__f__max_features=1, m__bagging_temperature=0.38292687475378984, m__depth=15, m__grow_policy=Lossguide, m__l2_leaf_reg=2, m__iterations=12, score=(train=-22.070, test=-20.141), total=   4.1s
[CV] f__f__max_features=1, m__bagging_temperature=0.38292687475378984, m__depth=15, m__grow_policy=Lossguide, m__l2_leaf_reg=2, m__iterations=12 
0:	learn: 7.6424938	total: 578us	remaining: 6.36ms
1:	learn: 7.5415389	total: 1.47ms	remaining: 7.37ms
2:	learn: 7.4452611	total: 2.17ms	remaining: 6.52ms
3:	learn: 7.3534848	total: 3.01ms	remaining: 6.01ms
4:	learn: 7.2660381	total: 3.66ms	remaining: 5.13ms
5:	learn: 7.1827533	total: 4.4ms	remaining: 4.4ms
6:	learn: 7.1034665	total: 5.11ms	remaining: 3.65ms
7:	learn: 7.0280179	total: 5.75ms	remaining: 2.88ms
8:	learn: 6.9562517	total: 6.47ms	remaining: 2.16ms
9:	learn: 6.8880162	total: 7.23ms	remaining: 1.45ms
10:	learn: 6.8231638	total: 7.92ms	remaining: 720us
11:	learn: 6.7615509	total: 8.53ms	remaining: 0us

Kernel died, restarting


 
Restarting kernel... 


 
[SpyderKernelApp] WARNING | No such comm: cba83eba3ca511eb87119cb6d01cd292
C:\Users\Thomas Wolf\anaconda3\envs\ChemPipeline\lib\site-packages\joblib\externals\loky\backend\resource_tracker.py:318: UserWarning: resource_tracker: There appear to be 970 leaked folder objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
C:\Users\Thomas Wolf\anaconda3\envs\ChemPipeline\lib\site-packages\joblib\externals\loky\backend\resource_tracker.py:333: UserWarning: resource_tracker: C:\Users\THOMAS~1\AppData\Local\Temp\joblib_memmapping_folder_23356_a8106b6619f048cc9af4ad0835ca3e3c_56a9164ead0d4cdb97d98ba6db8803ea: FileNotFoundError(2, 'Das System kann den angegebenen Pfad nicht finden')
  warnings.warn('resource_tracker: %s: %r' % (name, e))

**Comment by kizill** [comment]
Hello! I'm debugging this problem, will be fixed soon, i hope.

**Comment by kizill** [comment]
Fixes are in master - fixed empy trees inference in non-symmetric case. Added boost_from_average support for MultiRMSE.
Will appear in nearest release, thanks for your help!

**Comment by kizill** [comment]
Released in 0.24.4, please try it :)

**Comment by ThomasWolf0701** [comment]
> 
> 
> Released in 0.24.4, please try it :)

Thanks will give it a try. Is the newest version already available via conda/pip

**Comment by LyzhinIvan** [comment]
> Is the newest version already available via conda/pip
Yes, it is available via conda and pip.

**Comment by ThomasWolf0701** [comment]
> 
> 
> Released in 0.24.4, please try it :)

Works now ! :) Thanks for fixing this.

**Comment by ThomasWolf0701** [comment]
While it now works for training and classification , I now get a kernel crash when calculating SHAP values on the resulting model. Both with the SHAP package and the catboost internal SHAP calculation.

**Comment by ek-ak** [comment]
Thanks for trying! Could you provide a reproducing example? I suppose that the data remains the same as before?

**Comment by ThomasWolf0701** [comment]
> 
> 
> Thanks for trying! Could you provide a reproducing example? I suppose that the data remains the same as before?

https://github.com/catboost/catboost/issues/1568

**Comment by ThomasWolf0701** [comment]
> 
> 
> Thanks for trying! Could you provide a reproducing example? I suppose that the data remains the same as before?

Yes did not change the data.

**Comment by SebastianCallh** [comment]
Any updates on this? I am running into the exact same error when calling `get_feature_importance`.
